index.bs

```diff
@@ -387,233 +507,172 @@ Each \[[Stage]] value indicates the currently pending operation. If the \[[Resul
   </tr>
 </table>
 
-<h4 id="registry-GetRegistryEntry" aoid="GetRegistryEntry">GetRegistryEntry(registry, key)</h4>
+<h3 id="registry-entry-constructor">The Registry Entry Constructor</h3>
 
-The abstract operation GetRegistryEntry with arguments registry and key performs the following steps:
+The Registry Entry constructor is the %RegistryEntry% intrinsic object. When called as a constructor it creates and initializes a new %RegistryEntry% object. %RegistryEntry% is not intended to be called as a function and will throw an exception when called in that manner.
 
-<emu-alg>
-1. If Type(_registry_) is not Object, throw a *TypeError* exception.
-1. Let _entries_ be _registry_.[[RegistryData]].
-1. Let _pair_ be the entry in _entries_ such that _pair_.[[key]] is equal to _key_.
-1. If _pair_ does not exist, then return *null*.
-1. Let _entry_ be _pair_.[[value]].
-1. Let _currentStage_ be GetCurrentStage(_entry_).
-1. Let _result_ be the result of transforming _currentStage_.[[Result]] with a new pass-through promise.
-1. Let _stage_ be _currentStage_.[[Stage]].
-1. Let _result_ be CreateObject().
-1. Call SimpleDefine(_result_, "stage", _stage_).
-1. Call SimpleDefine(_result_, "result", _result_).
-1. If _stage_ is "ready" then let _module_ be _entry_.[[Module]].
-1. Else let _module_ be *undefined*.
-1. Call SimpleDefine(_result_, "module", _module_).
-1. If _entry_.[[Error]] is *nothing*, then:
-  1. Call SimpleDefine(_result_, "error", *null*).
-1. Else:
-  1. Let _opt_ be CreateObject().
-  1. Call SimpleDefine(_opt_, "value", _entry_.[[Error]]).
-  1. Call SimpleDefine(_result_, "error", _opt_).
-1. Return _result_.
-</emu-alg>
-
-A <dfn id="registry-entry">registry entry</dfn> is a record with the following fields:
```

Comment:
this was moved down to the properties of the registry entry instance, under internal slot subsection.